pkg/kubectl/cmd/run.go

```diff
@@ -413,7 +413,9 @@ func waitForPod(podClient coreclient.PodsGetter, ns, name string, exitCondition
 		ev, err := watch.Until(0, w, func(ev watch.Event) (bool, error) {
```

Comment:
No, `watch.Until()` does not return a `nil` event without an error (it can return both non-`nil` event and an error though). But the line that I replaced was trying to access `ev.Object` even in case if `err` is not nil.